### DISCUSSION TITLE
Running Cloudflare Workers  on Docker : can not access on http://localhost:8787

### DISCUSSION THREAD

**Discussion by gitadityakumar** [honojs/hono#2337] [General]
**Issue Description:**
Made a simple Hono app with Cloudflare Workers, dockerized it, and now unable to access it on http://localhost:8787.

**Docker Container Logs:**
```
$ docker run -p 8787:8787 honov1

> dev
> wrangler dev src/index.ts

 ⛅️ wrangler 3.32.0
-------------------
⎔ Starting local server...
[wrangler:inf] Ready on http://localhost:8787
```

**Possible Issues Checked:**
1. **Verify Port Mapping:** Ensure that the port mapping is correctly configured to map port 8787 of the container to port 8787 of the host machine.
2. **Firewall:** Check for any firewall rules that might be blocking incoming connections on port 8787.
3. **Application Configuration:** Ensure that the application inside the Docker container is correctly configured to listen on port 8787.
4. **Network Interface:** Confirm that Docker is using the correct network interface. If using Docker Desktop, it typically binds to localhost, but in other setups, it might bind to a different network interface.
5. **Restart Docker Service:** Restart the Docker service to see if it resolves any networking issues.

**Comment by sam-rusty** [comment]
did you find any solution to this problem?

**Comment by sam-rusty** [comment]
just found a workaround. adding --ip i am finally able to get it working
```
wrangler dev --ip=127.0.0.1
```

**Reply by gitadityakumar** [reply]
yeah but you need  to spacify the port attributes to . here is a complete dockerfile which is running a hello word dockerized app successfully 
------------------------------------------------------------------------------------------------------------------------------------------------------
# Use a Node.js base image
FROM node:18-slim

# Set working directory
WORKDIR /app

# Copy package files to install dependencies
COPY package.json package-lock.json ./

# Install dependencies
RUN npm install --production

# Install wrangler globally
RUN npm install -g wrangler@3.99.0

# Copy the rest of the application files
COPY . .

# Expose the port that wrangler will use
EXPOSE 8787

# Command to start the application
CMD ["wrangler", "dev", "--ip=0.0.0.0", "--port=8787"]
